README.md

```diff
@@ -432,21 +431,31 @@ Each `LogEntry__c` record automatically stores the component's type ('Aura' or '
 
 #### Example LWC Usage
 
-For lightning component developers, the `logger` LWC provides very similar functionality that is offered in Apex. Simply import the `logger` LWC in your component, and call the desired logging methods within your code.
+For lightning component developers, the `logger` LWC provides very similar functionality that is offered in Apex. Simply import the `getLogger` function in your component, use it to initialize an instance once per component, and call the desired logging methods within your code.
 
 ```javascript
 // For LWC, import logger's createLogger() function into your component
-import { createLogger } from 'c/logger';
+import { getLogger } from 'c/logger';
+import callSomeApexMethod from '@salesforce/apex/LoggerLWCDemoController.callSomeApexMethod';
 
-export default class LoggerLWCImportDemo extends LightningElement {
-  logger;
+export default class LoggerDemo extends LightningElement {
+  // Call getLogger() once per component
```

Comment:
What about something like this for `loggerService.js`:

```js
export function getLoggerService() {
  return !LoggerService.loggerService ? new LoggerService() : LoggerService.loggerService;
}

export default class LoggerService {
  static hasInitialized = false;
  static loggerService;
  // etc...
    constructor() {
    this._loadSettingsFromServer();

    if (areSystemMessagesEnabled && !LoggerService.hasInitialized) {
      this._logToConsole('INFO', 'logger component initialized\n' + JSON.stringify(new BrowserContext(), null, 2));

      LoggerService.hasInitialized = true;
      LoggerService.loggerService = this;
    } else if (!LoggerService.hasInitialized) {
      LoggerService.hasInitialized = true;
      LoggerService.loggerService = this;
    }
  }

   // etc ...
}
```

And then in `logger.js`, just swap out the import and call that function:

```js
export function getLogger() {
  return getLoggerService();
}
```

That way it doesn't matter how many times `getLogger()` is called

Reply:
I think at least for now, it shouldn't return a static instance of the service - I'm definitely open to changing some of this, but using static would ~break~ change the existing behavior. Since JavaScript/lightning components don't have a direct equivalent to a transaction in Apex/Flow, it's still helpful to have some kind of "boundary" in JS to group related log entries - so each component having a separate instance sort of serves as the "boundary", each `LoggerService` instance manages all of the entries for a particular component.

- Each instance of `LoggerService` has its own buffer of log entries, and when they're saved, they're all related to the same `Log__c` record. This makes it easier to find log entries created from `c/someComponent` vs `c/anotherComponent`
- With a static instance of `LoggerService`, any time 1 component calls `saveLog()`, log entries from all LWCs on the page would get saved (which could be nice) & then tied to a single `Log__c` (which changes reporting/could be confusing...?)

I definitely want to consider changing/improving how some of this works in JS (and there's a semi-related open issue #496), but changing it will impact some orgs (in terms of reporting, etc), so I want to hold off on doing so for the moment. But let's definitely chat about this more whenever we have a chance.

Reply:
Hmmm... at least for now, I could make the `#settings` variable in `LoggerService` a static variable - no matter how many instances of `LoggerService` there are at any point, the settings should be the same, so making it static makes sense, I think. That should cut that Apex controller call down to 1 call per page load (instead of 1 call per instance of `LoggerService`)